hoodie-hadoop-mr/src/test/java/com/uber/hoodie/hadoop/InputFormatTestUtil.java

```diff
@@ -116,20 +115,25 @@ public static File prepareParquetDataset(TemporaryFolder basePath, Schema schema
 
     }
 
-    private static Iterable<? extends GenericRecord> generateAvroRecords(Schema schema, int numberOfRecords, String commitTime) {
+    private static Iterable<? extends GenericRecord> generateAvroRecords(Schema schema, int numberOfRecords, String commitTime, String fileId) {
         List<GenericRecord> records = new ArrayList<>(numberOfRecords);
         for(int i=0;i<numberOfRecords;i++) {
-            records.add(generateAvroRecord(schema, i, commitTime));
+            records.add(generateAvroRecord(schema, i, commitTime, fileId));
         }
         return records;
     }
 
-    private static GenericRecord generateAvroRecord(Schema schema, int recordNumber,
-        String commitTime) {
-        return new GenericRecordBuilder(schema).set("field1", "field" + recordNumber)
-            .set("field2", "field" + recordNumber)
-            .set(HoodieRecord.COMMIT_TIME_METADATA_FIELD, commitTime)
-            .set(HoodieRecord.COMMIT_SEQNO_METADATA_FIELD, commitTime + "_" + recordNumber).build();
+    public static GenericRecord generateAvroRecord(Schema schema, int recordNumber,
+        String commitTime, String fileId) {
+        GenericRecord record = new GenericData.Record(schema);
+        record.put(HoodieRecord.COMMIT_TIME_METADATA_FIELD, commitTime);
+        record.put("field1", "field" + recordNumber);
+        record.put(HoodieRecord.RECORD_KEY_METADATA_FIELD, "key_" + recordNumber);
+        record.put("field2", "field" + recordNumber);
+        record.put(HoodieRecord.PARTITION_PATH_METADATA_FIELD, commitTime);
+        record.put(HoodieRecord.FILENAME_METADATA_FIELD, fileId);
+        record.put(HoodieRecord.COMMIT_SEQNO_METADATA_FIELD, commitTime + "_" + recordNumber);
+        return record;
     }
```

Comment:
So made this change along with adding sample-test-evolved.avro. The GenericRecordBuilder tries to set default values into the record for the fields which are not present (when creating the record). GenericData.Record ignores setting default values for the absent keys and so the schema is applied during read time (which is when the default values will be used).  
https://avro.apache.org/docs/1.7.7/api/java/org/apache/avro/generic/GenericRecordBuilder.html